### PR TITLE
[WIP] Allow compiler options in golang.

### PR DESCRIPTION
**Pull request by take-cheeze**
I want `-gcflags '-m'` flag to see escape and inline result.

### CODE REVIEW

File: cattleshed-conf/compilers.py [hunk 1]
```diff
@@ -1922,14 +1922,15 @@ def make_go(self):
...
-                'compiler-option-raw': False,
+                'compiler-option-raw': True,
                 'compile-command': ['/bin/true'],
                 'version-command': version_command,
                 'switches': [],
                 'initial-checked': [],
                 'display-name': display_name,
-                'display-compile-command': 'go run prog.go',
-                'run-command': ['/opt/wandbox/go-{cv}/bin/go', 'run', 'prog.go'],
+                'display-compile-command': ['go', 'build', '-o', 'prog', 'prog.go'],
```
Comment (melpon) [inline cattleshed-conf/compilers.py:1931]: `display-compile-command` must be string not array.

File: cattleshed-conf/compilers.py [hunk 2]
```diff
@@ -1922,14 +1922,15 @@ def make_go(self):
...
+                'compile-command': ['/opt/wandbox/go-{cv}/bin/go', 'build', '-o', 'prog', 'prog.go'],
+                'run-command': ['./prog'],
                 'runtime-option-raw': True,
```
Comment (melpon) [inline cattleshed-conf/compilers.py:1934]: `runtime-option-raw` should be `False` if `compiler-option-raw` is `True`

File: cattleshed-conf/compilers.py [hunk 3]
```diff
@@ -1922,14 +1922,15 @@ def make_go(self):
...
+                'compiler-option-raw': True,
                 'compile-command': ['/bin/true'],
                 'version-command': version_command,
                 'switches': [],
                 'initial-checked': [],
                 'display-name': display_name,
-                'display-compile-command': 'go run prog.go',
-                'run-command': ['/opt/wandbox/go-{cv}/bin/go', 'run', 'prog.go'],
+                'display-compile-command': ['go', 'build', '-o', 'prog', 'prog.go'],
+                'compile-command': ['/opt/wandbox/go-{cv}/bin/go', 'build', '-o', 'prog', 'prog.go'],
```
Comment (melpon) [inline cattleshed-conf/compilers.py:1932]: Compiler options can only be appended to the end of `compile-command`. So remove `prog.go`.

**Comment by melpon** [comment]
suggests:

```diff
--- a/cattleshed-conf/compilers.py
+++ b/cattleshed-conf/compilers.py
@@ -395,6 +395,10 @@ class Switches(object):
                 'flags': ['-package', 'core'],
                 'display-name': 'Jane Street Core',
             },
+            'go-gcflags-m': {
+                'flags': ['-gcflags', '-m'],
+                'display-name': "-gcflags '-m'",
+            },
         }
 
     def make(self):
@@ -1922,15 +1926,16 @@ class Compilers(object):
                 'displayable': True,
                 'language': 'Go',
                 'output-file': 'prog.go',
-                'compiler-option-raw': False,
+                'compiler-option-raw': True,
                 'compile-command': ['/bin/true'],
                 'version-command': version_command,
-                'switches': [],
+                'switches': ['go-gcflags-m'],
                 'initial-checked': [],
                 'display-name': display_name,
-                'display-compile-command': 'go run prog.go',
-                'run-command': ['/opt/wandbox/go-{cv}/bin/go', 'run', 'prog.go'],
-                'runtime-option-raw': True,
+                'display-compile-command': 'go build -o prog',
+                'compile-command': ['/opt/wandbox/go-{cv}/bin/go', 'build', '-o', 'prog'],
+                'run-command': ['./prog'],
+                'runtime-option-raw': False,
                 'jail-name': 'melpon2-default',
                 'templates': ['go'],
             }, cv=cv))
```

**Comment by take-cheeze** [comment]
@melpon 
Thank you for the review.

I've found that it still doesn't work since `go build` command's flags handling is different from others.
In `go build` command flags needs to be passed before input files so is it possible to add options like `compile-option-insert-position` ?
Seems I need to changes to:
https://github.com/melpon/wandbox/blob/f19098f12c86078a16016d524bbf7b8e21f4fb89/cattleshed/src/load_config.cc#L244

Though in switches `insert-position` seems to work:
https://github.com/melpon/wandbox-builder/blob/c7ba2be9f1161c58449c92bc3a34d4768b6e9f68/cattleshed-conf/compilers.py#L387

**Comment by take-cheeze** [comment]
And I found some display bug so reported: https://github.com/melpon/wandbox/issues/228

**Comment by melpon** [comment]
`go build` command does not require input files. So you may not use `insert-position` option.

**Comment by take-cheeze** [comment]
@melpon Totally missed the last review. Sorry.
The suggestion seems better so you can commit that instead of this PR.

**Comment by melpon** [comment]
Add `-gcflags '-m'` option https://wandbox.org/permlink/meHg93ixjPJcSUn7. Thank you for requesting!

**Comment by take-cheeze** [comment]
:+1: Thanks for implementing!

**Pull request closed without merging**